mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -879,6 +879,7 @@ object Matrices {
       case dm: BDM[Double] =>
         new DenseMatrix(dm.rows, dm.cols, dm.data, dm.isTranspose)
       case sm: BSM[Double] =>
+        sm.compact()
```

Comment:
We probably should not mutate the argument, unless we check all use cases and make sure this is OK.  I don't like the idea of copying unnecessarily, but it might be needed.

Alternatively, can we just send a fix to Breeze since the error is there?